modules/standard/SysError.chpl

```diff
@@ -52,6 +52,164 @@ class SystemError : Error {
   proc writeThis(f) {
     try! f.write(msg);
   }
+
+  proc type fromSyserr(err: syserr, msg: string = "") {
+    var strerror_err: err_t = ENOERR;
+    var errstr : c_string = sys_strerror_syserr_str(err, strerror_err);
+    var err_msg: string   = errstr:string;
+    if !msg.isEmptyString() then
+      err_msg = errstr:string + " (" + msg + ")";
```

Comment:
I see you are combining the msg argument with the result of strerror here. A reasonable alternative would be to create an overload of Error's proc message() in SystemError that does this combining. It's not strictly necessary to do it when creating the SystemError.